src/roles/wildchild.py

```diff
@@ -81,45 +72,42 @@ def on_myrole(evt, var, user):
 
 @event_listener("del_player")
 def on_del_player(evt, var, user, mainrole, allroles, death_triggers):
-    if var.PHASE not in var.GAME_PHASES:
-        return
-
     for child in get_all_players(("wild child",)):
-        if child.nick not in IDOLS or child in evt.params.deadlist or users._get(IDOLS[child.nick]) not in evt.params.deadlist: # FIXME
+        if child in evt.params.deadlist or IDOLS.get(child) not in evt.params.deadlist:
             continue
 
-        # change their main role to wolf, even if wild child was a template
-        child.send(messages["idol_died"])
-        WILD_CHILDREN.add(child.nick)
+        # Change their main role to wolf
+        child.send(messages["wild_child_idol_died"])
+        WILD_CHILDREN.add(child)
         change_role(child, get_main_role(child), "wolf")
         var.ROLES["wild child"].discard(child)
 
-        wcroles = var.WOLFCHAT_ROLES
         if var.RESTRICT_WOLFCHAT & var.RW_REM_NON_WOLVES:
             if var.RESTRICT_WOLFCHAT & var.RW_TRAITOR_NON_WOLF:
                 wcroles = var.WOLF_ROLES
             else:
                 wcroles = var.WOLF_ROLES | {"traitor"}
-        wolves = get_players(wcroles)
-        wolves.remove(child)
-        if wolves:
-            for wolf in wolves:
-                wolf.queue_message(messages["wild_child_as_wolf"].format(child))
-            wolf.send_messages()
-        if var.PHASE == "day":
-            random.shuffle(wolves)
-            new = []
-            for i, wolf in enumerate(wolves):
-                wolfroles = get_main_role(wolf)
-                cursed = ""
-                if wolf in get_all_players(("cursed villager",)):
-                    cursed = "cursed "
-                new.append("\u0002{0}\u0002 ({1}{2})".format(wolf, cursed, wolfroles))
-
-            if new:
-                child.send(messages["wolves_list"].format(", ".join(new)))
-            else:
-                child.send(messages["no_other_wolves"])
+
+            wolves = get_all_players(wcroles)
```

Comment:
get_players is correct here. People assigned wolfteam roles as secondary are not actual wolves and are thus not meant to be notified.

Reply:
Ok.